src/pioneer-p3dx/p3dx_dpl/src/motor_controler.cpp

```diff
@@ -52,24 +55,30 @@ int main(int argc, char **argv) {
     pub_left_ = argv_[1];
   }
   // set relative node namespace
-  ros::NodeHandle n("~");
-  n.getParam("tf_prefix", tf_prefix);
+  ros::NodeHandle n;
+  n.getParam("robot_id", robot_id);
 
+  // tf listener for calculation of wheel spacing
+  tf::TransformListener listener;
+  tf::StampedTransform transform;
   // get length in between wheels
   try {
-    tf::StampedTransform transform;
-    listener.lookupTransform(tf_prefix + "/base/joint1",
-                             tf_prefix + "/base/joint0", ros::Time(0),
-                             transform);
+    listener.waitForTransform(std::to_string(robot_id) + "/base/joint1",
+                              std::to_string(robot_id) + "/base/joint0",
+                              ros::Time::now(), ros::Duration(3.0));
+    listener.lookupTransform(std::to_string(robot_id) + "/base/joint1",
+                             std::to_string(robot_id) + "/base/joint0",
+                             ros::Time(0), transform);
     WHEELS_HALF_SPACING = transform.getOrigin().absolute().getY() / 2;
+    ROS_INFO("DPL: motor_controler WHEELS_HALF_SPACING: %f",
+             WHEELS_HALF_SPACING);
+
   } catch (tf::TransformException ex) {
     ROS_ERROR("Problem with transformation between wheels: %s", ex.what());
   }
 
-  // get wheel diameter
-
   // read msg from standard ROS topic for receiving Twist commands
-  n.subscribe("cmd_vel", 1000, callback_val);
+  ros::Subscriber twist_sub = n.subscribe("/cmd_vel", 1000, callback_val);
```

Comment:
It is iin need here. I am referencing something which is above my level in hierarchy. And as you eplained to me I can only look down and not to the same level or above. This is default value because most of the algiritms actualy work directly with /cmd_vel.  I am not sure how this algorithms behave when you have this topic in some namespace, but I hope it will be possible to change there. In addition in next release I will add option that you can pass this subsc. topic thru argv.